modules/openthread/platform/hdlc_interface.cpp

```diff
@@ -0,0 +1,404 @@
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+#include <assert.h>
+#include <openthread/tasklet.h>
+#include <openthread/platform/alarm-milli.h>
+#include <common/logging.hpp>
+#include <common/code_utils.hpp>
+#include "hdlc_interface.hpp"
+
+namespace ot {
+
+namespace Zephyr {
+
+HdlcInterface::HdlcInterface(const Url::Url &aRadioUrl)
+	: mEncoderBuffer()
+	, mHdlcEncoder(mEncoderBuffer)
+	, mHdlcRxCallbackField(nullptr)
+	, mReceiveFrameBuffer(nullptr)
+	, mReceiveFrameCallback(nullptr)
+	, mReceiveFrameContext(nullptr)
+	, mHdlcSpinelDecoder()
+	, mIsInitialized(false)
+	, mSavedFrame(nullptr)
+	, mSavedFrameLen(0)
+	, mIsSpinelBufferFull(false)
+	, mRadioUrl(aRadioUrl)
+{
+	mHdlcRxCallbackField = HdlcRxCallback;
+
+	radio_dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_hdlc_rcp_if));
+	__ASSERT_NO_MSG(device_is_ready(radio_dev));
+
+	hdlc_api = (struct hdlc_api *)radio_dev->api;
+	if (!hdlc_api)
+	{
+		otLogDebgPlat("Radio device initialization failed");
+		assert(0);
+	}
+}
+
+HdlcInterface::~HdlcInterface(void)
+{
+}
+
+otError HdlcInterface::Init(ReceiveFrameCallback aCallback, void *aCallbackContext, RxFrameBuffer &aFrameBuffer)
+{
+	int status;
+	otError error = OT_ERROR_NONE;
+
+	if (!mIsInitialized)
+	{
+		/* Event initialization */
+		k_event_init(&s_spinel_hdlc_event);
+
+		/* Read/Write semaphores initialization */
+		k_mutex_init(&s_spinel_hdlc_wr_lock);
+		k_mutex_init(&s_spinel_hdlc_rd_lock);
+
+		if (error == OT_ERROR_NONE)
+		{
+			/* Message queue initialization */
+			k_msgq_init(&spinel_hdlc_msgq, &spinel_hdlc_msgq_buffer, 1, 1);
+
+			mHdlcSpinelDecoder.Init(mRxSpinelFrameBuffer, HandleHdlcFrame, this);
+			mReceiveFrameCallback = aCallback;
+			mReceiveFrameContext = aCallbackContext;
+			mReceiveFrameBuffer = &aFrameBuffer;
+
+			/* Initialize the HDLC interface */
+			status = hdlc_api->register_rx_cb(mHdlcRxCallbackField, this);
+			if (status == 0)
+			{
+				mIsInitialized = true;
+			}
+			else
+			{
+				otLogDebgPlat("Failed to initialize HDLC interface %d", status);
+				error = OT_ERROR_FAILED;
+			}
+		}
+	}
+
+	return error;
+}
+
+void HdlcInterface::Deinit(void)
+{
+	int status;
+
+	status = hdlc_api->deinit();
+	if (status == 0)
+	{
+		mIsInitialized = false;
+	}
+	else
+	{
+		otLogDebgPlat("Failed to terminate HDLC interface %d", status);
+	}
```

Comment:
According to Zephyr CS, this should be:
```suggestion
	if (status == 0) {
		mIsInitialized = false;
	}  else {
		otLogDebgPlat("Failed to terminate HDLC interface %d", status);
	}
```
Please check in other places too.

Reply:
resolved